packaging/bundle-validation/validate.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# NOTE: this script runs inside hudi-ci-bundle-validation container
+# $WORKDIR/jars/ is supposed to be mounted to a host directory where bundle jars are placed
+# TODO: $JAR_COMBINATIONS should have different orders for different jars to detect class loading issues
+
+WORKDIR=/opt/bundle-validation
+HIVE_DATA=${WORKDIR}/data/hive
+JAR_DATA=${WORKDIR}/data/jars
+UTILITIES_DATA=${WORKDIR}/data/utilities
+
+run_hive_sync () {
+    echo "::warning::validate.sh setting up hive sync"
+    #put config files in correct place
+    cp $HIVE_DATA/spark-defaults.conf $SPARK_HOME/conf/
+    cp $HIVE_DATA/hive-site.xml $HIVE_HOME/conf/
+    ln -sf $HIVE_HOME/conf/hive-site.xml $SPARK_HOME/conf/hive-site.xml
+    cp $DERBY_HOME/lib/derbyclient.jar $SPARK_HOME/jars/
```

Comment:
interesting.. what is the consideration here that you moved these out from docker building? these configs are quite common to different bundle testing scenarios. if any specific setting, we can pass via CLI arg to the test case's spark command

Reply:
It was causing spark to try using the metastore and some strange behavior was happening. I am not sure if every one of those was causing the issue, but I just wanted to be safe.

Reply:
@jonvex in the latest refactoring commit i'm making shared conf for the whole stack, meaning the infra (hadoop, hive, etc) we setup should be appropriate to handle all bundle tests, no need to swap in and out some confs. We should be able to bypass the problem you encountered by doing direct spark datasource read (not via registered temp table)